base/errorshow.jl

```diff
@@ -111,8 +111,14 @@ end
 
 function showerror(io::IO, ex::LoadError, bt; backtrace=true)
     !isa(ex.error, LoadError) && print(io, "LoadError: ")
+    if bt isa StackTraces.StackTrace
+        # Remove the internals of how precompilation is set up
+        StackTraces.remove_frames!(bt, :include_package_for_output; reverse=true, n=1)
```

Comment:
This feels a bit error-prone - in particular, doesn't it require ruling out recursion in the loading code?

I wonder if `rethrow(truncate_bt=true)` / `throw(e; truncate_bt=true)` or `Base.truncate_catch_backtrace(...)` or a similar functionality would be better behaved

Reply:
Yeah I guess the most certain part of this PR is what the result should look like. How that's achieved I was less clear on a good design.

What would be a good test case for recursion?

Reply:
@topolarity if you had a particular loading pattern in mind it'd be great to understand it better.

Reply:
By recursion do you mean an included file that has an include in it?

Reply:
Bump

Reply:
Sorry for the delay here @IanButterworth, I've been sitting on some changes to our stack trace handling that I wanted to upstream so that we can do this properly (without scanning for function names, etc.)

Let me get those open and come back to this early next week 🙂

Reply:
Perfect. Thanks